### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by YgorSouza**
Hi,

I added some standard items to your crate docs, and fixed most clippy lints. Let me know if I got anything wrong.

You might want to add a CI workflow to check for compiler warnings or failing tests. I don't use GitHub much myself, but you can probably copy the one from [egui_dock](https://github.com/Adanos020/egui_dock), since it is very similar to your project.

### CODE REVIEW

**Comment by YgorSouza** [comment]
I also made some improvements to the `shift_vec` function [here](https://github.com/YgorSouza/egui_dnd/commit/04b38201180fd9b8fba5c7e710d6a8ce126c7df1) if you want to take a look. As far as I can tell it works the same as before. I can make another PR for that or add it to this one if you want.

File: src/lib.rs [hunk 1]
```diff
@@ -1,8 +1,33 @@
+#![doc = include_str!("../README.md")]
```
Comment (lucasmerlin) [inline src/lib.rs:1]: That's awesome!

**Review by lucasmerlin — APPROVED** [review]
Thanks for your PR! Really appreciated.

**Comment by lucasmerlin** [comment]
> I also made some improvements to the `shift_vec` function [here](https://github.com/YgorSouza/egui_dnd/commit/04b38201180fd9b8fba5c7e710d6a8ce126c7df1) if you want to take a look. As far as I can tell it works the same as before. I can make another PR for that or add it to this one if you want.

Looks good, feel free to make another PR!

**Pull request merged**